packages/@react-aria/grid/src/useGridRow.ts

```diff
@@ -49,7 +51,17 @@ export function useGridRow<T, C extends GridCollection<T>, S extends GridState<T
   });
 
   // TODO: move into useSelectableItem?
-  let {pressProps} = usePress({...itemProps, isDisabled});
+  let {pressProps, isPressed} = usePress({...itemProps, isDisabled});
+
+  // Get rid of click/drag/pointer/mousedown handlers if selection isn't allowed so that text selection can happen
+  if (!allowsSelection) {
+    pressProps = {
```

Comment:
This is relying on some implementation details of `usePress`. Maybe it would be better to pass through everything from `itemProps` except for `onPress`?

Reply:
Just to make sure I'm on the same page: this is suggesting that I modify `itemProps` if `!allowSelection` such that it doesn't have `onPress`/`onPressStart`/`onPressUp` when it gets passed to `usePress` correct?

Reply:
If so, `usePress` still returns `onPointerDown`/`onClick`/etc and those handlers do things like `preventDefault` and `disableTextSelection` which stop text selection from happening

Reply:
Would setting `isDisabled` on `usePress` when `allowsSelection` is false work?

Reply:
Unfortunately that doesn't work without modifications to `usePress`. If I change the following lines in `usePress` so that they early return if `isDisabled` then this appraoch will work, but I'm unsure what ramifications that would have on overall `usePress` behavior

https://github.com/adobe/react-spectrum/blob/main/packages/@react-aria/interactions/src/usePress.ts#L283
https://github.com/adobe/react-spectrum/blob/main/packages/@react-aria/interactions/src/usePress.ts#L318

Reply:
Yeah that's what I thought. That's why I originally suggested doing something like this:

```js
if (!allowsSelection) {
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  let {onPressStart, onPressEnd, onPressUp, onPress, ...otherProps} = itemProps;
  pressProps = otherProps;
}
```

We basically don't want to use the `pressProps` at all in that case, but only the other item props. I feel like this is less reliant on implementation details from `usePress` and `useSelectableItem` than specifically copying over certain DOM props. If either of those hooks changed in the future to add additional props the current solution would break.

Reply:
Ah I see, I misunderstood this earlier. Yeah this should work, I'll update